sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -153,15 +173,77 @@ private static BiFunction<HttpClientRequest, NettyOutbound, Publisher<Void>> bod
                     hdr.getValuesList().forEach(value -> reactorNettyRequest.addHeader(hdr.getName(), value));
                 }
             }
-            if (restRequest.getBody() != null) {
-                Flux<ByteBuf> nettyByteBufFlux = restRequest.getBody().map(Unpooled::wrappedBuffer);
-                return reactorNettyOutbound.send(nettyByteBufFlux);
+            BinaryData body = restRequest.getBodyAsBinaryData();
+            if (body != null) {
+                BinaryDataContent bodyContent = BinaryDataHelper.getContent(body);
+                if (bodyContent instanceof ByteArrayContent) {
+                    return reactorNettyOutbound.send(Mono.just(Unpooled.wrappedBuffer(bodyContent.toBytes())));
```

Comment:
Was there a reason why this uses the regular `send` API and not `sendByteArray`?

Reply:
This is a small optimization that showed up in benchmarks. I.e. it creates one layer of operators less than netty's impl.